src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8632,6 +8632,18 @@ public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOpera
             }
             VisitRvalue(node.Argument);
             // https://github.com/dotnet/roslyn/issues/31018: Check for delegate mismatch.
+            if (node.Argument.ConstantValue?.IsNull != true
+                && MakeMemberSlot(receiverOpt, @event) is not -1 and var memberSlot)
+            {
+                if (node.IsAddition)
+                {
+                    this.State[memberSlot] = this.State[memberSlot].Meet(ResultType.State);
+                }
+                else
+                {
+                    this.State[memberSlot] = NullableFlowState.MaybeNull;
+                }
```

Comment:
Perhaps use conditional: `this.State[memberSlot] = node.IsAddition ? ... : ...;`

Reply:
Done